code/game/objects/items/ashtray.dm

```diff
@@ -1,17 +1,31 @@
-/obj/item/ashtray
+/obj/item/storage/ashtray
 	icon = 'icons/ashtray.dmi'
-	var/max_butts = 0
+	w_class = WEIGHT_CLASS_SMALL
+	can_hold = list(/obj/item/cigbutt, /obj/item/clothing/mask/cigarette)
```

Comment:
based on line 27, shouldn't this except matches?

Reply:
Do roaches/rollies produce trash that would go in here as well?

Reply:
roaches/rollie butts are subclasses of /obj/item/cigbutt, does `can_hold` require exact types or can it infer what subtypes it supports?

Reply:
can hold appears to accept subtypes.